lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -111,6 +111,22 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
             objectKey, bucketMD, false, size, null, null, authInfo),
         log,
     };
+
+    // For Azure BlobStorage API compatability
+    // If an object already exists copy/repair creation-time
+    // creation-time must remain static after an object is created
+    //  --> EVEN FOR VERSIONS <--
+    if (objMD) {
+        if (objMD['creation-time']) {
```

Comment:
IMHO this is the kind of compat layer that could be done directly in `ObjectMD` class.

Reply:
That would be ideal, however there isn't really a place to put it where all ops touching MD would pass through. This area of the codebase is definitely in need of a refactor.